### ISSUE TITLE
Defining arrival rate per session?

### ISSUE THREAD

**Issue by robbles** [processone/tsung#156]
Is it possible to do this? The scenario I want to define has multiple types of sessions, each with their own arrival rate, and all arriving during the same phase (i.e. not separated out into their own phases).

According to the docs, you can set the probability of a given session in a phase:
```
<arrivalphase phase="3" duration="1" unit="minute">
  <session_setup name="http_test_1" probability="80"/>
  <session_setup name="fake"        probability="20"/>
  <users  interarrival="1" unit="second"/>
</arrivalphase>
```
But that doesn't seem to allow setting the arrivalrate separately. Is each user generated before a session is chosen?

**Comment by tisba** [comment]
I'm not quite sure, why you want to do that. Because with defining the distribution of the sessions, you are defining the arrival rate.

And yes, AFAIK, a user is generated and then a session is chosen, but again, I'm not sure, what the problem is you are trying to solve.

**Comment by robbles** [comment]
@tisba you're right - I realized this shortly before you replied. It's the same thing if I generate the overall arrival rate and then split up the traffic according to the distribution of session types.

The only reason I had to do it the other way round (generate session and arrival rate together) was to make it easier to increase the traffic from one session type without having to calculate the change to the global arrival rate.

Does that make sense? This way definitely works, so I'll be fine for now. But it seems like a more intuitive way of doing most tests would be to specify each session type with an arrival rate, rather than a global arrival rate and a session probability. Perhaps something like:

```xml
<arrivalphase phase="3" duration="1" unit="minute">
  <users  arrivalrate="10" unit="second" session="session_1" />
  <users  arrivalrate="4" unit="second" session="session_2" />
</arrivalphase>
```

**Comment by tisba** [comment]
Okay, I think I get it. It's an interesting idea. I think this should be doable, but not really straight forward currently.
